referenceframe/frame_system.go

```diff
@@ -397,6 +403,44 @@ func (sfs *simpleFrameSystem) getFrameToWorldTransform(inputMap map[string][]Inp
 	return srcToWorld, err
 }
 
+// ReplaceFrame finds the original frame which shares its name with replacementFrame. We then transfer the original
+// frame's children and parentage to replacementFrame. The original frame is removed entirely from the frame system.
+// replacementFrame is not allowed to exist within the frame system at the time of the call.
+func (sfs *simpleFrameSystem) ReplaceFrame(replacementFrame Frame) error {
+	var replaceMe Frame
+	if replaceMe = sfs.Frame(replacementFrame.Name()); replaceMe == nil {
+		return fmt.Errorf("%s not found in frame system", replacementFrame.Name())
+	} else if replaceMe == replacementFrame {
+		return fmt.Errorf("%s already exists in frame system", replacementFrame.Name())
+	}
+	if replaceMe == sfs.World() {
+		return fmt.Errorf("%s is the world of the frame system - which cannot be replaced", replaceMe.Name())
+	}
+
+	// get replaceMe's parent
+	replaceMeParent, err := sfs.Parent(replaceMe)
+	if err != nil {
+		return err
+	}
+
+	// remove replaceMe from the frame system
+	delete(sfs.frames, replaceMe.Name())
+	delete(sfs.parents, replaceMe)
+
+	for f, parent := range sfs.parents {
+		// replace frame with parent as replaceMe with replaceWith
+		if parent == replaceMe {
+			delete(sfs.parents, f)
+			sfs.parents[f] = replacementFrame
+		}
+	}
+	// add replacementFrame to frame system with parent of replaceMe
+	if err = sfs.AddFrame(replacementFrame, replaceMeParent); err != nil {
+		return err
+	}
+	return nil
```

Comment:
This all can collapse to `return sfs.AddFrame(replacementFrame, replaceMeParent)`